### PR TITLE
LINDA balance adjustment

### PR DESCRIPTION
**Pull request by Allfd**
[LINDA PR Video - Youtube](https://www.youtube.com/watch?v=hcsRSuiFtsM)
[Paradise Forum Post](https://nanotrasen.se/forum/topic/10219-atmospherics-changes/)

[Atmospherics changes](http://i.imgur.com/OrrPUAq.png)

This PR modifies the atmospherics controller to control both pipe networks and atmospheric machines in addition to its current role of handling LINDA.  The amount of energy released is increased which when combined with a decrease in the scheduled interval results in a speed increase of LINDA of approximately 5x.

An additional mechanic has been added, for pressure differentials above a specific limit (default of 15kpa) instead of pushing, LINDA will now throw the object.  Terror Spiders have a push differential of 50 kPa and a throw differential of 100 kPa.

The cost of Nanofrost has been decreased from 100 units of water to 50.

The multiplication of the toxins transfer to air of 5x for the flamethrower has been removed, along with the unused throw_amount variable and user interface.

Reinforced walls have had their damage capacity increased from 200 to 600.

Air alarms in scrubbing mode will move to off if they enter a danger (red) state.

Frost will need to be removed with heat.

A map edit replaces all volume pumps with regular pumps in atmospherics.

🆑Fox McCloud and Alffd
add: Merge all LINDA and atmos machine functions into the air controller subprocess.
add: Air alarms in scrubbing mode will shut off when alarm enters danger state. If this occurs simultaneously with a max2 temperature alarm panic siphon will automatically activate.
add: High air pressure differences will throw objects and knock over station species.
add: Add magpulse to syndicate and death squad borgs.
tweak: Change all volume pumps to regular pumps in atmospherics to accommodate new LINDA speeds.
tweak: Change LINDA interval from 2 seconds to 400 milliseconds.
tweak: Removed unused control from flamethrower.
tweak: Set flamethrower release equal to fuel used.
tweak: Increased Reinforced Wall HP from 200 to 600.
tweak: Buff aliens against space wind.
tweak: Buff Terror Spiders against space wind.
tweak: Frost requires heat to remove.
/🆑

Special thanks to @Anticept and @FlattestGuitar for early testing and feedback along with everyone else who participated, willing or otherwise.

Extra special thanks to @Aurorablade for figuring out the source of the crashes.

### CODE REVIEW

**Comment by taukausanake** [comment]
Watching Science burn like that, seeing the flames spread and melt everything, that was beautiful. It's also been a long time since I've seen canisters bust like that.

So you showed perma shooting someone out at a fast speed but what if they were on the other side of the room? I guess what I'm asking is does this mimic ZAS murder depressurization?

**Comment by Allfd** [comment]
It depends, a lot...

So if the hole is small, no, you will be fine on the other side of the room as the gradient that forms will not be harsh enough to eject you.  This is especially true if you are in a big room.

If the room is more like maintenance it is more likely.

It depends on both the layout and size of the room, and size and position of the hole.

**Comment by taukausanake** [comment]
It's a start I suppose. Now I wonder how many Science sabotage methods are going to use melting Toxin Storage

**Comment by BiberDark** [comment]
You are my hero.  YES .
I posted some ideas for more deadliness to your forum post.

**Comment by SamHPurp** [comment]
@taukausanake are you suggesting limiting access to toxin storage to the RD / those with Plasma Research job titles etc?

@Allfd @Fox-McCloud this looks fantastic. May be worth looking at buffing the Fire Fighter Mechs in response to this.

**Comment by Fox-McCloud** [comment]
Needs pics of the map change.

File: code/game/objects/items/weapons/shards.dm [hunk 1]
```diff
@@ -87,6 +87,7 @@
 				H.Weaken(3)
 				if(affecting.take_damage(5, 0))
 					H.UpdateDamageIcon()
+					affecting.embed(src)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/shards.dm]: Oh, uh, well. This should probably be its own PR; this doesn't really have anything to do with LINDA, at all.

Comment (SamHPurp) [inline code/game/objects/items/weapons/shards.dm]: Agreed, I like the change though, but it probably shouldn't be a 100% chance

**Comment by Ralta** [comment]
As a query relating to the glass shards, will it affect you regardless of what you're wearing? i.e. will you still get a shard embedded in your foot if you're wearing a hardsuit or similar protective gear?

**Comment by SamHPurp** [comment]
@Ralta It'll be only if you aren't wearing shoes for whatever reason.

**Comment by FreeStylaLT** [comment]
I'm fully in support of the concept, though, I am unsure if the specific increases should stay, as from what I have gathered regarding the CPU difference, this would keep us unlisted, generally, even more than we already are.

We will likely have to adjust either the unlisting thresholds or the PR numbers to not keep us from being likely perma-unlisted.

**Comment by SamHPurp** [comment]
@FreeStylaLT if thats the case I'd be happy to see Goonlights go to remedy that.

**Comment by taukausanake** [comment]
I don't think I understand why the CPU doing more work would require an unlisting, unless you mean the number of people online would push on it more with x5 LINDA. I rather not see Goonlights go in any case.

@SamHPurp No of course not. The access for Toxins storage is fine as far as I know. What I was saying was anyone with access could melt Science by doing exactly what was done in testing. Honestly, where did you get that from when I revel in the possibility for death by natural causes and sabotage? :^)

**Comment by BiberDark** [comment]
If you have to choose between lights and air . Goodbye lights.

File: code/game/objects/items/weapons/shards.dm (hunk 1 above)
Comment (taukausanake) [inline code/game/objects/items/weapons/shards.dm]: Glass getting stuck in your foot? I'd say 50-80%. People make broken glass sound so dangerous that you shouldn't even touch it with bare hands so stepping on it should be bad

Comment (Allfd) [inline code/game/objects/items/weapons/shards.dm]: I put this in at the last minute to address a edge case, if you are throw. Into space with a piece of glass, you will be ejected at the same speed.

This means that each turf you cross triggers the stepping on glass.

Comment (taukausanake) [inline code/game/objects/items/weapons/shards.dm]: Is there a way to remove stepping on anything when there is no gravity? Is that not the case already, I don't know

**Comment by Allfd** [comment]
The effects on CPU are hard to quantify as it is very round type dependent, the speed of this PR was lowered from 10x because of this.

On an active atmospherics round this means about 100 players is ideal, on an inactive atmospherics round it has negligible CPU effect.

**Comment by taukausanake** [comment]
I think this warrants a test run with a full population just to see how it goes. It's not going to hurt to push this to the server for a round or two just to see how it runs will it? At the very worse what's a little bit of downtime due to server reverting?

**Comment by Allfd** [comment]
This test has been run a few times now over the last week.

**Comment by taukausanake** [comment]
Wait really? I thought that was something else entirely? Just cranking up the timer for LINDA actions or whatever. So this is the thing that was tested when that spaceworm ate the station? Any maybe when any and every other explosion recently?

**Comment by Krausus** [comment]
> I don't think I understand why the CPU doing more work would require an unlisting

Allfd has the server set up to unlist during periods of high CPU usage, because CPU usage sorta-kinda correlates to active player count. This may raise the question, "why not unlist based on player count?", and we *do,* but lower CPU usage is better, and correlation is sorta-kinda like causation, so unlisting ≈ fewer players ≈ less CPU usage. (Full disclosure: I was opposed to this when I heard about it, but was willing to give it a try on the basis that we could disable it if it became a problem)

> from what I have gathered regarding the CPU difference, this would keep us unlisted

We can solve this by not unlisting based on CPU usage.

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 2]
```diff
@@ -7594,12 +7594,12 @@
...
 "cQh" = (/obj/machinery/meter{frequency = 1443; id = "dloop_atm_meter"; name = "Distribution Loop"},/obj/machinery/atmospherics/pipe/manifold/visible/cyan,/turf/simulated/floor/plasteel,/area/atmos/distribution)
 "cQi" = (/obj/machinery/atmospherics/pipe/simple/visible/cyan{dir = 10; initialize_directions = 10; level = 2},/obj/machinery/alarm{frequency = 1439; pixel_y = 23},/turf/simulated/floor/plasteel,/area/atmos/distribution)
-"cQj" = (/obj/machinery/atmospherics/binary/volume_pump/on{dir = 8; name = "Air To Distro"},/obj/structure/cable{icon_state = "0-2"; pixel_y = 1; d2 = 2},/obj/machinery/atmospherics/pipe/simple/hidden/supply,/obj/machinery/power/apc{dir = 1; name = "north bump Engineering"; pixel_x = 0; pixel_y = 24; shock_proof = 1},/turf/simulated/floor/plasteel,/area/atmos/distribution)
+"cQj" = (/obj/machinery/atmospherics/binary/pump{dir = 8; name = "Air To Distro"; on = 1},/obj/structure/cable{icon_state = "0-2"; pixel_y = 1; d2 = 2},/obj/machinery/atmospherics/pipe/simple/hidden/supply,/obj/machinery/power/apc{dir = 1; name = "north bump Engineering"; pixel_x = 0; pixel_y = 24; shock_proof = 1},/turf/simulated/floor/plasteel,/area/atmos/distribution)
```
Comment (SamHPurp) [inline _maps/map_files/cyberiad/cyberiad.dmm]: Should probably update Metastation too, or I could just put this into my next batch of fixes.

**Comment by Anticept** [comment]
Would this help with CPU prioritization and lag? https://github.com/tgstation/tgstation/pull/17987

**Comment by alexkar598** [comment]
I don't think having two CL for differents authors works well

**Comment by Anticept** [comment]
Here's what i have for quickly setting up airlocks in emergencies that should be implemented soon. Put a digital valve into an inflatable, and now it has cycleable valves. Link two of them together, and toggling one toggles the other.

As shown:

https://cdn.discordapp.com/attachments/145700045008797697/309411263237324800/unknown.png

**Comment by Regen1** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by scrubmcnoob** [comment]
calm down google @Regen1 
This is looking good.
No more hug box space.

File: _maps/map_files/cyberiad/cyberiad.dmm (hunk 2 above)
Comment (Fox-McCloud) [inline _maps/map_files/cyberiad/cyberiad.dmm]: Metastation doesn't use volume pumps in atmospherics.

**Comment by Spacemanspark** [comment]
Regens is very happy today.

**Comment by taukausanake** [comment]
Hey does this mean we'll need cycling airlocks now? Sounds _wonderful_. Maybe we should work on a better implementation for them once this is merged

File: code/game/objects/items/weapons/shards.dm (hunk 1 above)
Comment (Aurorablade) [inline code/game/objects/items/weapons/shards.dm]: i would forgive it for being an edge case.

Comment (taukausanake) [inline code/game/objects/items/weapons/shards.dm]: RIP feet. And maybe RIP off in pieces if enough damage. Sounds like fun

Comment (Allfd) [inline code/game/objects/items/weapons/shards.dm]: I will put into its own PR.

However, unless merged, if you are ejected with a glass shard and are barefoot, it will do damage to you with each turf you travel up to the damage cap of your mob.

**Comment by Vivalas** [comment]
Yes cycling airlocks are much better than magical tiny fans.

File: code/LINDA/LINDA_turf_tile.dm [hunk 3]
```diff
@@ -305,17 +305,25 @@
...
-		if(pressure_difference > pressure_resistance)
+		if(pressure_difference >= throw_pressure_limit)
+			var/general_direction = get_edge_target_turf(src,direction)
+			if(ismob(src))
+				to_chat(src, "<span class='userdanger'>The pressure sends you flying!</span>")
+			spawn throw_at(general_direction,pressure_difference/10,pressure_difference/200,src,1)
 			last_forced_movement = air_master.current_cycle
+			return 1
+		else if(pressure_difference > pressure_resistance)
 			spawn step(src, direction)
```
Comment (Allfd) [inline code/LINDA/LINDA_turf_tile.dm]: It does, if these are not called its possible for the controller to hang

File: code/LINDA/LINDA_turf_tile.dm [hunk 4]
```diff
@@ -305,17 +305,25 @@
...
 /atom/movable/proc/experience_pressure_difference(pressure_difference, direction)
 	if(last_forced_movement >= air_master.current_cycle)
 		return 0
 	else if(!anchored && !pulledby)
-		if(pressure_difference > pressure_resistance)
+		if(pressure_difference >= throw_pressure_limit)
+			var/general_direction = get_edge_target_turf(src,direction)
+			if(ismob(src))
+				to_chat(src, "<span class='userdanger'>The pressure sends you flying!</span>")
+			spawn throw_at(general_direction,pressure_difference/10,pressure_difference/200,src,1)
```
Comment (Allfd) [inline code/LINDA/LINDA_turf_tile.dm]: I couldn't find it in the coding guidelines but I made the change anyways.

File: code/LINDA/LINDA_turf_tile.dm [hunk 5]
```diff
@@ -305,17 +305,24 @@
...
 /atom/movable/var/last_forced_movement = 0
 
 /atom/movable/proc/experience_pressure_difference(pressure_difference, direction)
 	if(last_forced_movement >= air_master.current_cycle)
 		return 0
 	else if(!anchored && !pulledby)
-		if(pressure_difference > pressure_resistance)
+		if(pressure_difference >= throw_pressure_limit)
+			var/general_direction = get_edge_target_turf(src,direction)
+			if(ismob(src)) to_chat(src, "<span class='userdanger'>The pressure sends you flying!</span>")
```
Comment (tigercat2000) [inline code/LINDA/LINDA_turf_tile.dm]: Please don't do one-line conditionals.

File: code/LINDA/LINDA_turf_tile.dm [hunk 6]
```diff
@@ -305,17 +305,24 @@
...
+/atom/movable/var/throw_pressure_limit = 15
 /atom/movable/var/last_forced_movement = 0
 
 /atom/movable/proc/experience_pressure_difference(pressure_difference, direction)
 	if(last_forced_movement >= air_master.current_cycle)
 		return 0
 	else if(!anchored && !pulledby)
-		if(pressure_difference > pressure_resistance)
+		if(pressure_difference >= throw_pressure_limit)
+			var/general_direction = get_edge_target_turf(src,direction)
```
Comment (tigercat2000) [inline code/LINDA/LINDA_turf_tile.dm]: Spaces between arguments.

File: code/LINDA/LINDA_turf_tile.dm [hunk 7]
```diff
@@ -305,17 +305,24 @@
...
+			if(ismob(src)) to_chat(src, "<span class='userdanger'>The pressure sends you flying!</span>")
+			throw_at_fast(general_direction,pressure_difference/10,pressure_difference/200,src,1)
```
Comment (tigercat2000) [inline code/LINDA/LINDA_turf_tile.dm]: Spaces between operators.

File: code/LINDA/LINDA_turf_tile.dm (hunk 5 above)
Comment (Allfd) [inline code/LINDA/LINDA_turf_tile.dm]: I thought this is want @Krausus wanted me to do.

Comment (tigercat2000) [inline code/LINDA/LINDA_turf_tile.dm]: @Allfd the opposite, actually, you were using `spawn` in-line, he wanted it to be ```
spawn
   throw_at()
```

Comment (Krausus) [inline code/LINDA/LINDA_turf_tile.dm]: Ah, I worded that poorly, didn't I? By "one-line code block", I meant the spawn's block of code being on the same, one line as the `spawn` statement. This:
```
spawn throw_at(...)
```
should've looked like this (if you hadn't replaced it with `throw_at_fast`):
```
spawn()
  throw_at(...)
```
The `spawn step(...)` below does the same thing; it makes it less clear that it's spawning off execution of a separate block of code.

**Comment by Allfd** [comment]
@tigercat2000 @Krausus changes done.

**Comment by KingPhilipIII** [comment]
Now maybe as a malfunctioning AI I can flood plasma and actually have it do shit.

**Comment by granodd** [comment]
I'm not sure how I feel about this.

**Comment by DesolateG** [comment]
"A map edit replaces all volume pumps with regular pumps in atmospherics." 

Curious why you did this?

**Comment by Allfd** [comment]
@DesolateG 

Um...  Sending to @Fox-McCloud

**Comment by Allfd** [comment]
@Fox-McCloud @Crazylemon64 
DNM

bug with pipenet.

File: code/LINDA/LINDA_turf_tile.dm [hunk 8]
```diff
@@ -207,7 +207,23 @@
 				if(excited_group)
 					last_share_check()
 
-	air.react()
+	if(air)  //How did this even break?
```
Comment (tigercat2000) [inline code/LINDA/LINDA_turf_tile.dm]: Please do not do this. LINDA intentionally does not have null protection in many parts of it- This *should* runtime, because it should never happen.

Comment (Allfd) [inline code/LINDA/LINDA_turf_tile.dm]: I understand where you are coming from, but it seems to happen a great deal and it can bring down the server as happened twice in two rounds.

image

Comment (Fox-McCloud) [inline code/LINDA/LINDA_turf_tile.dm]: Could you explain what that chart means? Also, what leads you to believe that this, specifically, is bringing the server down?

I think the point Tigercat is trying to make is that unless there's definitive proof that this is what is truly bringing the server down, we want this to runtime, as air being `null` on a turf is indicative of a much larger problem, and the code here is masking the problem (why air is null to begin with).

It's kinda like if a human mob had a `null` species; if that ever happens, it shouldn't be painted over with a `if(species)` check, but rather the underlying problem of why a human mob isn't getting a species to begin with.

**Comment by Allfd** [comment]
So the graph is the count of runtimes from this bug since January of this year, the reason I think this is bringing down the server, is that its the last runtime to occur before the process controller hangs and we have to abandon the round.

File: code/LINDA/LINDA_turf_tile.dm (hunk 8 above)
Comment (Allfd) [inline code/LINDA/LINDA_turf_tile.dm]: After talking it over with Fox, this can't be causing the issues I thought it was,

Going to pull that commit.

**Comment by Allfd** [comment]
@Fox-McCloud @Crazylemon64 

Additional testing failed to reveal the bug I thought existed, removing the WIP label.

**Comment by Deanthelis** [comment]
Nuke ops should have free magboots in their lockers again, then. Otherwise when they break into anywhere from outside they'll be violently flung about all the time.

They'll still do that, of course, but it'll be because they forgot their boots, like the good ol' days.

**Comment by SamHPurp** [comment]
@Deanthelis #7268 does exactly this.

**Comment by adamkad1** [comment]
this is wrong, whenver i go into pressurized tile from space, it knocks me over, it doesnt work that way does it?

**Comment by Vivalas** [comment]
Outdated review, merge when I want my hardcore atmos.

**Comment by ProperPants** [comment]
@Vivalas your comment would be better noticed if you @tigercat2000 since it's their review keeping this back at the moment

**Comment by Allfd** [comment]
I made a temporary fix for the airlock to buy time for @SamHPurp to fix their merge conflict and deconflict the two PRs.

image

**Comment by Bram961** [comment]
Because of the faster speed ice that appears on the floor has a tendency to flash in some cases. it just rapidly appears and then immediately disappears again as the pressure reaches 0. this is especially noticeable on the bridge with shutters on the north west side of meta station when it's opened under pressure.

File: code/LINDA/LINDA_turf_tile.dm [hunk 9]
```diff
@@ -221,7 +221,7 @@
 
 	if(air.temperature < T0C && air.return_pressure() > 10)
 		icy = 1
-	else
+	else if(air.temperature > T0C)
```
Comment (Allfd) [inline code/LINDA/LINDA_turf_tile.dm]: @Bram961 Corrected with a logic change.

File: code/modules/mob/living/simple_animal/hostile/alien.dm [hunk 10]
```diff
@@ -23,8 +23,8 @@
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
 	unsuitable_atmos_damage = 15
 	heat_damage_per_tick = 20
-	pressure_resistance = 50    //50 kPa difference required to push
-	throw_pressure_limit = 100  //100 kPa difference required to throw
+	pressure_resistance = 100    //100 kPa difference required to push
+	throw_pressure_limit = 120   //120 kPa difference required to throw
```
Comment (SamHPurp) [inline code/modules/mob/living/simple_animal/hostile/alien.dm]: It was pretty silly being stuck in space as a xeno queen. Felt like a right lemon. 👍

Comment (MarcellusPye) [inline code/modules/mob/living/simple_animal/hostile/alien.dm]: Aww, no more spacing the xenos to victory?

File: code/game/machinery/alarm.dm [hunk 11]
```diff
@@ -302,11 +302,18 @@
 
 	if(old_danger_level!=danger_level)
 		apply_danger_level()
+		if(mode == AALARM_MODE_SCRUBBING && danger_level == ATMOS_ALARM_DANGER)
+			if(pressure_dangerlevel == ATMOS_ALARM_DANGER)
+				mode = AALARM_MODE_OFF
+				if(temperature_dangerlevel == ATMOS_ALARM_DANGER && cur_tlv.max2 <= environment.temperature)
```
Comment (MarcellusPye) [inline code/game/machinery/alarm.dm]: I'ed put this outside of if(mode == AALARM_MODE_SCRUBBING), as a atmos tech I like to set toxins to contaminated at round start, to save time 😄

**Comment by Jovaniph** [comment]
Right so after playing with Fastmos for a while as an Atmospheric Tech. I came to the conclusion that the station isn't ready for it. Not for a while at least. There needs a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot a lot

of improvements... I don't expect this PR to merge any time soon and maybe not for a year or two.

**Comment by SamHPurp** [comment]
@Jovaniph Perhaps detail some of these as you see them.

**Comment by outbools** [comment]
@SamHPurp let me detail fastmos for ya:
every gas leak causes pressure to throw you and fuck you up. Throw n2o or plasma grenade and you not only flood room with gas - you also throw everyone into nearest wall. Traitor easy mode massacre was never that easier. And this will fuck up everything in all such situations - is it yellow warning sign on atmos alert? oh just 90% of normal atmosphere this is fi-- and that's where you got thrown down the hallway... and then thrown back.
All away missions with pressurized surface now a wild ride when you got thrown into xeno nest by atmos.
Muh lavaland is even worst - you fly directly into lava. Cool then.

and also i show this to one of ZAS devs and that's what he said:
ZAS dev: that's not how gas works
ZAS dev: it just shows "I PUT THE NUMBER HIGHER"
ZAS dev: Para is just trying to get LINDA to be superior to ZAS but it'll never happen at this rate

**Comment by Jovaniph** [comment]
I never played with ZAS before but I heard it takes the area of the room instead of a single tile. I would think ZAS is more realistic because for one, how many of you blown up a balloon before? Did the air expand to one side of the balloon or is it expanding in all directions? (rhetorical question). That's how air works.

**Comment by taukausanake** [comment]
ZAS operated with areas, but the issue was that in a pressurized area, think normal air even, any new gas introduced into it would be spread evenly through the area without any prior mixing. You can open a can of plasma in one corner of a room and the whole room would be filled immediately. That's the only real problem ZAS had but it was so minor compared to all the murder bonuses you got from it that it was pretty much ignored.
Also note that the R-Ust engine couldn't work without that issue and that is a _nice_ engine

**Comment by SamHPurp** [comment]
@outbools did a ZAS dev really just try to claim that ZAS is more realistic?

ZAS where if I fart on one side of the bridge, and the door is open on the other, everyone in the Box ring around it is going to be smelling my fart instantly.

**Comment by DesolateG** [comment]
@SamHPurp  don't forget temperature.

**Comment by Anticept** [comment]
ZAS is great where pressure is a concern. That's where it is realistic and I don't really see linda ever beating it in this regard anytime soon.

Linda is great on everything else.

**Comment by Allfd** [comment]
@outbools 
I am a huge fan of ZAS, it is not realistic to move back to ZAS however.

LINDA pressure is a function of a number of different components, including the moles of gas on a tile, and its temperature.  It is a tile by tile calculation.  ZAS effects an entire area.

From a technical perspective, this means that the smallest unit of calculation for pressure is 2 meters cubed.  The default throw value is 15 kPa which translates into 2.18 PSI.  The smallest unit of calculation we have is 2 meters squared.  That is 3100 inches squared

By multiplying 2.18 by 3100, we get a little over 6750 pounds of force being exerted.  @outbools Which would open a final question, is your character able to push a 6750 object?

These numbers get worse when you talk about a hull breach, instead of 15kPa we have a 101.3kPa change.

This is roughly 48,000 pounds of force in our 2 meter squared hole measure, or 14.69 psi.  Which by definition would be traveling at the speed of sound due to vacuum.

To put this into perspective, the Manhattan project scientists and US military considered a 5 psi overpreassure to be a 100% fatality area for a nuclear blast.

I have attached a link of a home being subjected to a 5psi differential.

https://en.wikipedia.org/wiki/Little_Boy#/media/File:House_1953_Nevada_Nuclear_Test_5_psi.jpg

Magboots are actually really amazing if you think about it.

**Comment by Anticept** [comment]
> Magboots are actually really amazing if you think about it.

As are people's ankles.

**Comment by SamHPurp** [comment]
> > Magboots are actually really amazing if you think about it.

> As are people's ankles.

It would be unfun, but I still like the idea of the pressure being too high, with magboots... you lose your feet.

**Comment by outbools** [comment]
@Allfd 
adjusted **LINDA**:
Man enters a room - steps three tiles further - get thrown. _"Invisible de/overpressurized throwing tiles"_ is not something you would consider as realistic or fun to interact with. 
Your PR was played for a month on my downstream server with ~20 players average and LINDA-time calculated pressure is what players point as unplayable.

**Comment by outbools** [comment]
Oh i forgot to mention locker throwing is too random and inconsistent

**Comment by adamkad1** [comment]
Plz remove this trash

**Comment by SamHPurp** [comment]
Agreed. Please remove @adamkad1 from talking on Github.

**Comment by adamkad1** [comment]
atleast you dont get screwed over by this

**Comment by mrturkeytoe** [comment]
Now we just need spesswind to stun. :+1:

**Comment by taukausanake** [comment]
@mrturkeytoe No, we need slamming into a wall to stun. I haven't been keeping up with this though so I don't know if that's in

**Comment by mrturkeytoe** [comment]
@taukausanake In ZAS pressure differentials AND wall slams stun, juicy wambo combo

**Comment by Allfd** [comment]
This has been open since may, no activity in months.  Closing.

**Comment by taukausanake** [comment]
RIP FUCKING EVERYTHING
